test/libyul/yulOptimizerTests/commonSwitchCasePrefixMover/nested.yul

```diff
@@ -0,0 +1,26 @@
+{
+  let a := calldataload(42)
+  switch a
+  case 0 {
+    let b := calldataload(23)
+    switch b
+    case 0 { sstore(0, 1) }
+    default { sstore(0, 1) }
+  }
+  default { sstore(1, 2) }
+}
```

Comment:
How about a case where a switch can be simplified only after simplifying nested switches?

```
{
    let a := calldataload(42)
    switch a
    case 0 {
        let b := calldataload(42)
        switch b
        case 0 { sstore(0, 1) }
        default { sstore(0, 1) }
    }
    default {
        let c := calldataload(42)
        sstore(0, 1)
        switch c
        case 0 {}
        default {}
    }
}
```